src/marshal.rs

```diff
@@ -0,0 +1,68 @@
+use crate::conversion::AsPyPointer;
+use crate::ffi;
+use crate::{PyErr, PyObject, PyResult, Python};
+use std::os::raw::c_int;
+
+/// The current version of the marshal binary format.
+pub const VERSION: i32 = 4;
+
+/// Serialize an object to bytes using the Python built-in marshal module.
+///
+/// The built-in marshalling only supports a limited range of object.
+/// See the [python documentation](https://docs.python.org/3/library/marshal.html) for more details.
+pub fn dumps(py: Python, object: &PyObject, version: i32) -> PyResult<Vec<u8>> {
```

Comment:
How about returning `PyResult<&PyBytes>`?
E.g.,
```rust
use crate::conversion::FromPyPointer;
...
FromPyPointer::from_owned_ptr_or_err(py, bytes)
```
would work.

Reply:
Ah, that avoids the possibly pointless copy into a `Vec`. Nice :)